pkg/cmd/cli/cmd/set/env.go

```diff
@@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
 	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
 	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
+	configMapName := kcmdutil.GetFlagString(cmd, "from-configmap")
+	prefix := kcmdutil.GetFlagString(cmd, "prefix")
```

Comment:
plus what's the argument against allowing it?  having --prefix be a universal flag seems better than making it linked to other flags, as long as there's an intelligent action we can take when it's specified.  (only argument against it i can think of is increasing test surface area and overall feature complexity)